QuestionFour/QuestionFour/Program.cs

```diff
@@ -6,10 +6,151 @@
 
 namespace QuestionFour
 {
-    class Program
+    public class Program
     {
         static void Main(string[] args)
         {
+            double a = 0;
```

Comment:
The code in your main program is much too long.  You should refactor into a separate function(s) that the main program calls.